cpp/oneapi/dal/backend/primitives/distance/correlation_distance_dpc.cpp

```diff
@@ -0,0 +1,111 @@
+template <ndorder order>
+auto distance<Float, correlation_metric<Float>>::get_deviation(const ndview<Float, 2, order>& inp,
+                                                               const event_vector& deps) const
+    -> comp_dev_res_t {
+    return compute_deviation(q_, inp, deps);
+}
+
+template <typename Float>
+template <ndorder order1, ndorder order2>
+sycl::event distance<Float, correlation_metric<Float>>::operator()(
+    const ndview<Float, 2, order1>& inp1,
+    const ndview<Float, 2, order2>& inp2,
+    ndview<Float, 2>& out,
+    const ndview<Float, 1>& inp1_norms,
+    const ndview<Float, 1>& inp2_norms,
+    const event_vector& deps) const {
+    auto ip_event = compute_cosine_inner_product(q_, inp1, inp2, out, deps);
+    return finalize_cosine(q_, inp1_norms, inp2_norms, out, { ip_event });
+}
+
+template <typename Float>
+template <ndorder order1, ndorder order2>
+sycl::event distance<Float, correlation_metric<Float>>::operator()(
+    const ndview<Float, 2, order1>& inp1,
+    const ndview<Float, 2, order2>& inp2,
+    ndview<Float, 2>& out,
+    const event_vector& deps) const {
+    auto [centered_inp1, comp_dev1_event] = get_deviation(inp1, deps);
+    auto [centered_inp2, comp_dev2_event] = get_deviation(inp2, deps);
+
+    auto [inv_norms1_array, inv_norms1_event] =
+        get_inversed_norms(centered_inp1, { comp_dev1_event });
+    auto [inv_norms2_array, inv_norms2_event] =
+        get_inversed_norms(centered_inp2, { comp_dev2_event });
+    return this->operator()(centered_inp1,
+                            centered_inp2,
+                            out,
+                            inv_norms1_array,
+                            inv_norms2_array,
+                            { inv_norms1_event, inv_norms2_event });
+}
+
+#define INSTANTIATE(F, A, B)                                                                    \
+    template sycl::event distance<F, correlation_metric<F>>::operator()(const ndview<F, 2, A>&, \
+                                                                        const ndview<F, 2, B>&, \
+                                                                        ndview<F, 2>&,          \
+                                                                        const ndview<F, 1>&,    \
+                                                                        const ndview<F, 1>&,    \
+                                                                        const event_vector&)    \
+        const;                                                                                  \
+    template sycl::event distance<F, correlation_metric<F>>::operator()(const ndview<F, 2, A>&, \
+                                                                        const ndview<F, 2, B>&, \
+                                                                        ndview<F, 2>&,          \
+                                                                        const event_vector&)    \
```

Comment:
please, delete \ to have const on the same line with const event_vector&)

Reply:
I keep applying your suggestion but `clang-format` just puts it back to this form. Not sure what I'm doing wrong.